management_endpoint.bal

```diff
@@ -0,0 +1,230 @@
+// Copyright (c) 2021 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+import ballerina/lang.'xml as xmllib;
+import ballerina/stringutils;
+
+# Eventhub management client implementation.
+#
+# + config - Client configuration
+public client class ManagementClient {
+
+    private ClientEndpointConfiguration config;
+    private string API_PREFIX = EMPTY_STRING;
+    private http:Client clientEndpoint;
+
+    public function init(ClientEndpointConfiguration config) {
+        self.config = config;
+        self.API_PREFIX = TIME_OUT + config.timeout.toString() + API_VERSION + config.apiVersion;
+        self.clientEndpoint = new (HTTPS + self.config.resourceUri);
+    }
+
+    # Create a new Eventhub
+    #
+    # + eventHubPath - event hub path
+    # + eventHubDescription - event hub description
+    # + return - Return XML or Error
+    remote function createEventHub(string eventHubPath, EventHubDescription eventHubDescription = {}) 
+            returns @tainted xml|error {
+        http:Request req = getAuthorizedRequest(self.config);
+        xmllib:Element eventHubDes = <xmllib:Element> xml `<EventHubDescription 
+            xmlns:i="http://www.w3.org/2001/XMLSchema-instance" 
+            xmlns="http://schemas.microsoft.com/netservices/2010/10/servicebus/connect"/>`;
+        req.setXmlPayload(getDescriptionProperties(eventHubDescription, eventHubDes));
+        string requestPath = FORWARD_SLASH + eventHubPath + self.API_PREFIX;
+        http:Response response = <http:Response> check self.clientEndpoint->put(requestPath, req);
+        if (response.statusCode == SUCCESS) {
+            xml xmlPayload = check response.getXmlPayload();
+            return xmlPayload;
+        }
+        return getErrorMessage(response);
+    }
+
+    # Get Eventhub description
+    #
+    # + eventHubPath - event hub path
+    # + return - Return XML or Error
+    remote function getEventHub(string eventHubPath) returns @tainted xml|error {
+        http:Request req = getAuthorizedRequest(self.config);
+        string requestPath = FORWARD_SLASH + eventHubPath;
+        http:Response response = <http:Response> check self.clientEndpoint->get(requestPath, req);
+        if (response.statusCode == OK) {
```

Comment:
Should we compare the status code constants in http package?

Reply:
Fixed. Thank you.